target-platform-configuration/src/main/java/org/eclipse/tycho/target/TargetPlatformConfigurationMojo.java

```diff
@@ -192,6 +192,13 @@ public class TargetPlatformConfigurationMojo extends AbstractMojo {
     @Parameter(name = DefaultTargetPlatformConfigurationReader.TARGET_DEFINITION_INCLUDE_SOURCE)
     private IncludeSourceMode targetDefinionIncludeSource;
 
+    /**
+     * When <code>true</code>, the resolved target platform is augmented with matching source
+     * bundles that can be found in the dependencies sources.
+     */
+    @Parameter(name = DefaultTargetPlatformConfigurationReader.INCLUDE_ALL_SOURCES)
+    private boolean includeAllSources;
```

Comment:
Should not
https://tycho.eclipseprojects.io/doc/latest/target-platform-configuration/target-platform-configuration-mojo.html#targetDefinitionIncludeSource 

with `targetDefinitionIncludeSource=force`

be something you like to archive here? In such case all locations are instructed to include the source bundles regardless of configuration in the target platform.